enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java

```diff
@@ -159,4 +148,114 @@ void testConvertUiRiskFlagsXCF() throws Exception {
         final PreApprovalScreen preApprovalScreen = ui.getPreApprovalScreen();
         assertEquals(PreApprovalScreen.ScreenType.WARNING, preApprovalScreen.getType());
     }
+
+    @Test
+    void testConvertUiPostApprovalMerchantRedirect() throws Exception {
+        final OperationDetailResponse operationDetail = createOperationDetailResponse();
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+        operationTemplate.setUi("{\n" +
+                "  \"postApprovalScreen\": {\n" +
```

Comment:
I realize that the original proposal was to have polymorphism even for `postApprovalScreen` itself, but it does not make sense to me yet. We may extend it later. More important is to have generic payload, which you may see below.

Reply:
This is not necessary right now as far as I can tell.

Reply:
@romanstrobl In fact, I was trying to say that I had not implemented what you mentioned at https://github.com/wultra/enrollment-server/pull/591#pullrequestreview-1229232076

Reply:
@kober32 Do you need a discriminator even for `postApprovalScreen` or is enough only for `payload`?
```json
"postApprovalScreen": {
   "type": "REVIEW",
```

Reply:
AFAIK both `heading` and `message` should be always required.

Reply:
I updated the comment to be more obvious that it is only a snippet.